### PR TITLE
improve comments for pattern, rename isPatternExpr

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### rename isPatternExpr

We define "pattern expression" as the entire pattern tree, which makes
isPatternExpr a misnomer. Renaming to `isPatternLeafOrTree`.

### CODE REVIEW

**Comment by RaduBerinde** [comment]
Maybe isPatternSentinel instead of isPatternLeafOrTree?

File: v3/pattern.go [hunk 1]
```diff
@@ -1,34 +1,70 @@
...
+//      |
+//      |-- pattern tree
+//
+//    Note the use of "pattern tree" for the ON conditions.
 
 var patternLeaf = &expr{}
 var patternTree = &expr{}
 
-func isPatternExpr(pattern *expr) bool {
+func isPatternLeafOrTree(pattern *expr) bool {
```
Comment (petermattis) [inline v3/pattern.go]: Yeah, I'd prefer `isPatternSentinel`.

File: v3/pattern.go [hunk 2]
```diff
@@ -1,34 +1,70 @@
...
-//     2: pattern tree
-//   1: pattern leaf
-//   2: pattern tree
+//  A Pattern Leaf matches any expression tree, with only the root of the tree
+//  being retained in a binding. It is used when the expression is used opaquely
+//  by the transformation. In other words, the transformation doesn't care
+//  what's inside the subtree. It is a "leaf" in the sense that it's a leaf in
+//  any binding matching a pattern.
+//
+//  1. Pattern Tree
```
Comment (petermattis) [inline v3/pattern.go]: `s/1/2/g`. Or perhaps get rid of the numbers.

Comment (RaduBerinde) [reply v3/pattern.go]: Removed the numbers

File: v3/pattern.go (hunk 1 above)
Comment (RaduBerinde) [reply v3/pattern.go]: Done.

**Pull request merged**